### PR TITLE
feat(core): added scrollable component

### PR DESCRIPTION
**Pull request by manyx**

### CODE REVIEW

**Review by vardidzel — CHANGES_REQUESTED** [review]
Please add `idea` files in `.gitignore`

File: core/src/components/cat-scrollable/cat-scrollable.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+})
+export class CatScrollable {
+
+  /** Flags to enable/disable scroll shadows. */
+  @Prop()
+  shadow: { x?: boolean; y?: boolean; } = { x: true, y: true };
+
+  /** Flags to enable/disable overflow. */
+  @Prop()
+  overflow: { x?: boolean; y?: boolean; } = { y: true };
```
Comment (vardidzel) [inline core/src/components/cat-scrollable/cat-scrollable.tsx:16]: Please move this  `{ x?: boolean; y?: boolean; }`  in another file.

File: core/src/components/cat-scrollable/cat-scrollable.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+@Component({
+  tag: 'cat-scrollable',
+  styleUrl: 'cat-scrollable.css',
+  shadow: true,
+})
+export class CatScrollable {
+
+  /** Flags to enable/disable scroll shadows. */
+  @Prop()
+  shadow: { x?: boolean; y?: boolean; } = { x: true, y: true };
```
Comment (vardidzel) [inline core/src/components/cat-scrollable/cat-scrollable.tsx:12]: same here

Comment (fynnfeldpausch) [reply core/src/components/cat-scrollable/cat-scrollable.tsx:12]: as far as I read in best practices, it is discouraged to use complex objects in properties. So I suggest to go with individual props like shadowX and shadowY

File: core/src/components/cat-scrollable/cat-scrollable.tsx (hunk 1 above)
Comment (fynnfeldpausch) [reply core/src/components/cat-scrollable/cat-scrollable.tsx:16]: same here

**Pull request merged**